recursion/core/src/multi/mod.rs

```diff
@@ -218,6 +254,8 @@ where
         let mut sub_builder = MultiBuilder::new(
             builder,
             local_multi_cols.is_poseidon2.into(),
+            local_multi_cols.poseidon2_first_row.into(),
+            builder.is_last_row(),
```

Comment:
Should we also add the condition `local_multi_cols.is_poseidon` (to handle the case when there are no poseidon2 rows).